components/post_view/index.js

```diff
@@ -36,14 +36,21 @@ const isChannelLoading = (params, channel, team, teammate) => {
 function makeMapStateToProps() {
     return function mapStateToProps(state, ownProps) {
         const team = getTeamByName(state, ownProps.match.params.team);
+        let channelLoading = true;
         let teammate;
         const channel = getChannel(state, ownProps.channelId);
+        let lastViewedAt = state.views.channel.lastChannelViewTime[ownProps.channelId];
+        if (channel) {
+            if (channel.type === Constants.DM_CHANNEL && channel.teammate_id) {
+                teammate = getUser(state, channel.teammate_id);
+            }
+            lastViewedAt = channel.last_post_at ? lastViewedAt : channel.last_post_at;
+        }
 
-        if (channel && channel.type === Constants.DM_CHANNEL && channel.teammate_id) {
-            teammate = getUser(state, channel.teammate_id);
+        if (getTeamMemberships(state)[team.id]) {
```

Comment:
If team is not part of memberships then we unMount the post_list. If user joins a new team we wait for membership again before mounting the children and calling the API calls for posts